### PR TITLE
Add users form with runtime joi validation

### PR DESCRIPTION
**Pull request by krakamod**

### CODE REVIEW

File: src/components/forms/input.jsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  static defaultProps = {
+    error: '',
+    type: 'text',
+    value: '',
+    onChange: null,
+  };
+
+  render() {
+    const {
+      name, label, error, ...rest
```
Comment (vik-buchinski) [inline src/components/forms/input.jsx]: don't use rest operator here

File: src/components/forms/select.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+class Select extends PureComponent<Props> {
+  static defaultProps = {
+    error: '',
+    value: 0,
+    onChange: null,
+  };
+
+  render() {
+    const {
+      name, label, options, error, ...rest
```
Comment (vik-buchinski) [inline src/components/forms/select.jsx]: remove rest parameter

File: src/containers/users/userForm.jsx [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    return currentTarget.value;
+  };
+
+  setErrors = (errors: Errors) => {
+    this.setState({ errors: errors || {} });
+  };
+
+  doSubmit = () => {
+    const { userId } = this.props;
+    if (userId === 'new') saveUser(this.state.user);
```
Comment (vik-buchinski) [inline src/containers/users/userForm.jsx]: Use curly brackets in if/else statements everywhere

File: src/containers/users/userForm.jsx [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+  validateProperty = ({ name, value }: HTMLInputElement) => {
+    const obj = { [name]: value };
+    const schema = { [name]: [bossSchema, employeeSchema][this.state.user.type][name] };
+    const { error } = Joi.validate(obj, schema);
+    return error ? error.details[0].message : null;
+  };
+
+  handleSelect = (event: SyntheticInputEvent<HTMLInputElement>) => {
+    const { currentTarget: input } = event;
+    this.setState((state) => {
```
Comment (vik-buchinski) [inline src/containers/users/userForm.jsx]: Please, avoid using function callback in setState. It is could grow to callback hell...
Just create object which you need to update in handleSelect function, and then:
`this.setState({ user }):`
And that it

File: src/components/forms/input.jsx (hunk 1 above)
Comment (krakamod) [reply src/components/forms/input.jsx]: Removed

File: src/components/forms/select.jsx (hunk 2 above)
Comment (krakamod) [reply src/components/forms/select.jsx]: Removed

File: src/containers/users/userForm.jsx (hunk 3 above)
Comment (krakamod) [reply src/containers/users/userForm.jsx]: Fixed

File: src/styles/components/buttons/button.js [hunk 5]
```diff
@@ -8,6 +8,9 @@ const Button: ComponentType<{
   nomargin?: boolean,
   right?: boolean,
   bold?: boolean,
+  big?: boolean,
```
[the remark this answers is not in the document]
Comment (krakamod) [reply src/styles/components/buttons/button.js]: Add string type of size

File: src/containers/users/userForm.jsx [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+        />
+        <Select
+          name="type"
+          label="Type"
+          value={user.type}
+          error={errors.type}
+          options={[{ id: 0, label: 'Boss' }, { id: 1, label: 'Employee' }]}
+          onChange={this.handleSelect}
+        />
+        {user.type === 0 && (
```
[the remark this answers is not in the document]
Comment (krakamod) [reply src/containers/users/userForm.jsx]: Moved

**Pull request merged**